### PR TITLE
PF-2411: rename terms for location/regions/datacenters

### PR DESCRIPTION
**Pull request by yuhuyoyo**
Region - a cloud platform region. What is now called a "datacenter" in TPS lingo
Zone - region with its zone, as Matthew suggests. We don't have anything specific in the API now that uses zone. It is an important part of the vocabulary that we can use in code comments, etc.
Location - a value in the ontology: it might be:
Geographic Location - Europe, Iowa
Region Location - us-central1

### CODE REVIEW

File: common/openapi.yml [hunk 1]
```diff
@@ -280,21 +280,21 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
-  /api/policy/v1alpha1/region:
+  /api/policy/v1alpha1/location:
     parameters:
       - $ref: '#/components/parameters/Platform'
-      - $ref: '#/components/parameters/RegionName'
+      - $ref: '#/components/parameters/GeographicLocation'
```
Comment (ddietterich) [inline common/openapi.yml:286]: I think you can use a region location here also, right? So this should not be specific to Geo?

File: common/openapi.yml [hunk 2]
```diff
@@ -773,23 +773,23 @@ components:
...
           name:
-            description: The region name
+            description: The geographic location name (e.g. europe, iowa)
             type: string
           description:
-            description: The region description
+            description: The geographic location description (e.g. Europe, Iowa)
             type: string
-          regions:
+          locations:
```
Comment (ddietterich) [inline common/openapi.yml:786]: nit: should description say "sub-location" instead of "subregion"?

File: common/openapi.yml [hunk 3]
```diff
@@ -429,10 +429,10 @@ components:
         type: string
         enum: ["gcp", "azure"]
 
-    RegionName:
-      name: region
+    GeographicLocation:
```
Comment (ddietterich) [inline common/openapi.yml:432]: I am thinking this should be just `Location`

File: service/src/main/java/bio/terra/policy/app/controller/ConversionUtils.java [hunk 4]
```diff
@@ -97,22 +97,21 @@ static ApiTpsPaoGetResult paoToApi(Pao pao) {
         .deleted((pao.getDeleted()));
   }
 
-  static ApiTpsRegion regionToApi(Region region) {
-    ApiTpsRegion apiRegion =
-        new ApiTpsRegion().name(region.getName()).description(region.getDescription());
+  static ApiTpsLocation regionToApi(Location region) {
+    ApiTpsLocation apiRegion =
```
Comment (ddietterich) [inline service/src/main/java/bio/terra/policy/app/controller/ConversionUtils.java:101]: Is it worth cleaning up the variable names, do you think?

**Review by ddietterich — COMMENTED** [review]
Main comment is maybe not to use GeographicLocation when the item can be `us-central1`.
I'm content to incrementally cleanup the variable names. wdyt?

File: service/src/main/java/bio/terra/policy/service/region/RegionService.java [hunk 5]
```diff
@@ -26,47 +27,48 @@
 public class RegionService {
   private static final String TERRA_REGION_CONSTRAINT = "terra:region-constraint";
   private static final String TERRA_REGION_ATTRIBUTE_NAME = "region-name";
-  private static final String GLOBAL_REGION = "global";
+  private static final String GLOBAL = "global";
```
Comment (mikenev) [inline service/src/main/java/bio/terra/policy/service/region/RegionService.java:30]: nit: Can we use GLOBAL_LOCATION? Having a GLOBAL makes sense for the value but it makes me think too much - like is it a global variable? Why do we have GLOBALS, etc.

**Review by ddietterich — APPROVED** [review]
lgtm

**Pull request merged**